#!/bin/sh

# this is a simple shell script to normalize ChIP-seq profiles
# inputs are bam files aligned to D. melanogaster (dmel)
# it counts the number of reads
# the number of reads is used as scaler
# bedgraphs are generated by bedtools

for BAM in *.dmel.bam; do

    BASE=`echo $BAM | sed 's/.dmel.bam//'`

    TOTAL=`samtools view -c ${BASE}.dmel.bam`
    SCALER=`echo "scale=10; 1/(${TOTAL}/1000000)" | bc`

    genomeCoverageBed -ibam ${BASE}.dmel.bam -fs 150 -bg -scale $SCALER  > ${BASE}.dmel_norm_to_dmel.bedgraph

done
